lib/trie/node.go

```diff
@@ -83,19 +84,19 @@ func (l *leaf) setGeneration(generation uint64) {
 }
 
 func (b *branch) copy() node {
-	b.Lock()
-	defer b.Unlock()
+	b.RLock()
+	defer b.RUnlock()
+
 	cpy := &branch{
 		key:        make([]byte, len(b.key)),
-		children:   [16]node{},
+		children:   b.children, // copy interface pointers
```

Comment:
if one of the copy's children changes, will that also change the child of the original node? if so seems kinda dangerous, but maybe go doesn't work like that? haha

Reply:
It's an array, not a slice :wink: 

```go
arr1 := [1]int{9}
arr2 = arr1
arr2[0] = 3
fmt.Println(arr1) // prints [9] still
```

aka array are always passed by value, there is no pointer unlike slices.

Reply:
oh yes it is an array! ok cool :D